fluent-langneg/src/negotiate_languages.ts

```diff
@@ -63,38 +48,32 @@ function GetOption(options, property, type, values, fallback) {
  *
  *     This strategy requires defaultLocale option to be set.
  */
-export default function negotiateLanguages(
-  requestedLocales,
-  availableLocales,
-  options = {}
-) {
-
-  const defaultLocale = GetOption(options, "defaultLocale", "string");
-  const strategy = GetOption(options, "strategy", "string",
-    ["filtering", "matching", "lookup"], "filtering");
-
-  if (strategy === "lookup" && !defaultLocale) {
-    throw new Error("defaultLocale cannot be undefined for strategy `lookup`");
-  }
-
-  const resolvedReqLoc = Array.from(Object(requestedLocales)).map(loc => {
-    return String(loc);
-  });
-  const resolvedAvailLoc = Array.from(Object(availableLocales)).map(loc => {
-    return String(loc);
-  });
+export function negotiateLanguages(
+  requestedLocales: Array<string>,
+  availableLocales: Array<string>,
+  {
+    strategy = "filtering",
+    defaultLocale,
+  }: NegotiateLanguagesOptions = {}
+): Array<string> {
 
   const supportedLocales = filterMatches(
-    resolvedReqLoc,
-    resolvedAvailLoc, strategy
+    Array.from(Object(requestedLocales)).map(String),
+    Array.from(Object(availableLocales)).map(String),
```

Comment:
I'm tempted to remove the string and array normalization here as well, because again, I think we can make reasonable assumptions about the quality of the input. There are tests which fail if I do that, however.